### PR TITLE
CORE-555: increase test coverage

### PR DESCRIPTION
**Pull request by robwtelicent**
Adding more tests. Also some refactoring in `TokenizerABAC.java` to break very large methods into smaller methods and consolidate some functionality, also various bits of unused code commented out pending review. Same in `Token.java`.

### CODE REVIEW

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java [hunk 1]
```diff
@@ -239,18 +211,18 @@ private Token parseToken() {
...
-            case CH_GT:         return maybeTwoChar(CH_GT, '=', TokenType.GT, ">", TokenType.GE, ">=");
+            case CH_EQUALS:     return maybeTwoChar(CH_EQUALS, CH_EQUALS, TokenType.EQ, TokenType.EQUIVALENT, "==");
+            case CH_EMARK:      return maybeTwoChar(CH_EMARK, CH_EQUALS, TokenType.EMARK, TokenType.NE, "!=");
+            case CH_LT:         return maybeTwoChar(CH_LT, CH_EQUALS, TokenType.LT, TokenType.LE, "<=");
+            case CH_GT:         return maybeTwoChar(CH_GT, CH_EQUALS, TokenType.GT, TokenType.GE, ">=");
 
-            case CH_VBAR:       return maybeTwoChar(CH_VBAR, '|', TokenType.VBAR, "|", TokenType.LOGICAL_OR, "||");
-            case CH_AMPHERSAND: return maybeTwoChar(CH_AMPHERSAND, '&', TokenType.AMPERSAND, "&", TokenType.LOGICAL_AND, "&&");
+            case CH_VBAR:       return maybeTwoChar(CH_VBAR, CH_VBAR, TokenType.VBAR, TokenType.LOGICAL_OR, "||");
+            case CH_AMPHERSAND: return maybeTwoChar(CH_AMPHERSAND, CH_AMPHERSAND, TokenType.AMPERSAND, TokenType.LOGICAL_AND, "&&");
```
Comment (rvesse) [inline rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java:220]: Random niggle - isn't `ampersand` misspelt here?  There shouldn't be a `h` in it should there?

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java [hunk 2]
```diff
@@ -319,106 +329,116 @@ private Token maybeTwoChar(char ch1, char secondChar, TokenType oneCharToken, St
...
+//        // 32 bit checks.
+//        return
+//            range(ch, 0x10000, 0x1FFFD) || range(ch, 0x20000, 0x2FFFD) || range(ch, 0x30000, 0x3FFFD) ||
+//            range(ch, 0x40000, 0x4FFFD) || range(ch, 0x50000, 0x5FFFD) || range(ch, 0x60000, 0x6FFFD) ||
+//            range(ch, 0x70000, 0x7FFFD) || range(ch, 0x80000, 0x8FFFD) || range(ch, 0x90000, 0x9FFFD) ||
+//            range(ch, 0xA0000, 0xAFFFD) || range(ch, 0xB0000, 0xBFFFD) || range(ch, 0xC0000, 0xCFFFD) ||
+//            range(ch, 0xD0000, 0xDFFFD) || range(ch, 0xE1000, 0xEFFFD);
+//    }
+
+// UNUSED
```
Comment (rvesse) [inline rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java:424]: Yeah so I think at one point Andy was considering potentially allowing attribute names and values to use IRIs, i.e. going very much all in on the Semantic Web

But we never went down that route as it made the labels excessively verbose.  I don't think we'd ever bring this particular capability back so may be best to remove it?

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java [hunk 3]
```diff
@@ -519,7 +538,7 @@ private void readNumber() {
                 token.setType(TokenType.HEX);
                 return;
             }
-        } else if ( ch == '-' || ch == '+' ) {
+        } else if ( ch == CH_MINUS || ch == CH_PLUS ) { // unreachable code?
```
Comment (rvesse) [inline rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java:541]: Yeah it looks like the `parseNumeric()` method (which you also cleaned up in this PR) already handles the `+` or `-` character prior to calling this method, so unless this method is called from anywhere else I think this branch is probably defunct

**Review by rvesse — APPROVED** [review]
Looks like a really nice tidy up

Added some comments about the status of some of the unreachable/commented out code blocks but don't need to be addressed for this PR necessarily

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java (hunk 1 above)
Comment (robwtelicent) [reply rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java:220]: You're right! Unfortunately this comes from `jena-base` so out of our control here. I could raise a PR against Jena if you think it is likely to get approved?

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java (hunk 3 above)
Comment (robwtelicent) [reply rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java]: Thanks for confirming. Removed.

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java (hunk 2 above)
Comment (robwtelicent) [reply rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java]: Now removed. If we ever did decide to bring it back we can always dig it up from the git history.

**Review by rvesse — APPROVED** [review]

File: rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java (hunk 1 above)
Comment (rvesse) [reply rdf-abac-core/src/main/java/io/telicent/jena/abac/attributes/syntax/tokens/TokenizerABAC.java:220]: Probably not worth it, it's such a low level thing that most people aren't going to look at that code so doesn't really matter if the constant is misspelt

**Pull request merged**